test/e2e/framework/kubelet/kubelet_pods.go

```diff
@@ -17,12 +17,23 @@ limitations under the License.
 package kubelet
 
 import (
+	"fmt"
+
 	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clientset "k8s.io/client-go/kubernetes"
-	"k8s.io/kubernetes/pkg/master/ports"
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+// GetKubeletPort find the kubelet port from the given nodeName.
+func GetKubeletPort(c clientset.Interface, nodeName string) (int, error) {
+	node, err := c.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return -1, fmt.Errorf("Failed to get node %s: %v", nodeName, err)
+	}
+	return int(node.Status.DaemonEndpoints.KubeletEndpoint.Port), nil
```

Comment:
Why is the casting here necessary, what is the returned type?

Reply:
@timothysc original type is `int32`